bin/lib/invalidUsernames.js

```diff
@@ -0,0 +1,160 @@
+const fs = require('fs-extra')
+const Handlebars = require('handlebars')
+const path = require('path')
+const { URL } = require('url')
+const util = require('util')
+
+const { loadConfig } = require('./common')
+const { isValidUsername } = require('../../lib/common/user-utils')
+const blacklistService = require('../../lib/services/blacklist-service')
+const { initConfigDir, initTemplateDirs } = require('../../lib/server-config')
+const { fromServerConfig } = require('../../lib/models/oidc-manager')
+
+const AccountManager = require('../../lib/models/account-manager')
+const EmailService = require('../../lib/services/email-service')
+const LDP = require('../../lib/ldp')
+const SolidHost = require('../../lib/models/solid-host')
+
+const fileExists = util.promisify(fs.exists)
+const fileRename = util.promisify(fs.rename)
+
+module.exports = function (program) {
+  program
+    .command('invalidusernames')
+    .option('--notify', 'Will notify users with usernames that are invalid')
+    .option('--delete', 'Will delete users with usernames that are invalid')
+    .description('Manage usernames that are invalid')
+    .action(async (options) => {
```

Comment:
My personal view: I move the implementation of the action into another file / function.
This way I can have different interfaces for it (CLI, Web UI etc).
Plus, it is easier to test.
Plus, it is a separation of concerns (CLI vs. using the lib).

Reply:
I would do it 100% if I thought this code was reusable, which I'm leaning toward it isn't =/

As I mentioned earlier, I found it difficult to write tests for this. It relies on a lot of file-based functionality, which makes it unsuited for unit tests. I could write integration tests, but didn't see any for the other commands, so didn't know how to attack it.

Reply:
From an Angular point of view I'd argue with Dependency Injection.
That is, providing the `fs` as an argument to the function.
But it feels meh somehow.

Reply:
I'm not that familiar with dependency injection in Node. I've tried [MockRequire](https://www.npmjs.com/package/mockrequire) earlier, but my tests always turned a nightmare to maintain.